server/service/src/preference/mod.rs

```diff
@@ -19,23 +19,28 @@ pub trait PreferenceServiceTrait: Sync + Send {
 
     fn get_preference_descriptions(
         &self,
-        connection: &StorageConnection,
+        connection: StorageConnection,
         store_id: Option<String>,
         pref_type: PreferenceType,
     ) -> Result<Vec<PreferenceDescription>, PreferenceError> {
         let PreferenceProvider {
             show_contact_tracing,
+            display_vaccine_in_doses,
         } = self.get_preference_provider();
 
-        let all_prefs_descriptions = vec![
-            // Add each pref here
-            show_contact_tracing.as_description(connection, store_id)?,
-        ];
+        let input = AppendIfTypeInputs {
+            pref_type,
+            connection,
+            store_id: store_id.clone(),
+        };
 
-        Ok(all_prefs_descriptions
-            .into_iter()
-            .filter(|pref| pref.preference_type == pref_type)
-            .collect())
+        let mut descriptions: Vec<PreferenceDescription> = Vec::new();
+
+        // Add each pref here
+        append_if_type(show_contact_tracing, &mut descriptions, &input)?;
+        append_if_type(display_vaccine_in_doses, &mut descriptions, &input)?;
```

Comment:
Bit of a change, before was loading all prefs, then filtering. Shouldn't have to provide storeId to load all the store prefs, if I actually just want the global ones. 

So, check if pref type is correct, then push into response vec (ideally would have a vec of the prefs, filter and then map, but because of Preference trait, it accepts <Value> generic, to put in a vec requires boxing any, type loss... seemed cleaner this way?)